test/e2e/lib/browser-manager.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * External dependencies
+ */
+import config from 'config';
+import playwright from 'playwright';
+
+const browserStartTimeoutMS = 2000;
+
+/**
+ * Returns the target screen size for tests to run against.
+ *
+ * By default, this function will return 'desktop' as the target.
+ * To specify another screen size, set the BROWSERSIZE environment variable.
+ *
+ * @returns {string} String representation of the target screen size.
+ */
+export function targetScreenSize() {
+	return ! process.env.BROWSERSIZE ? 'desktop' : process.env.BROWSERSIZE.toLowerCase();
+}
+
+/**
+ * Returns the locale under test.
+ *
+ * By default, this function will return 'en' as the locale.
+ * To set the locale, set the BROWSERLOCALE environment variable.
+ *
+ * @returns {string} String representation of the locale.
+ */
+export function targetLocale() {
+	return ! process.env.BROWSERLOCALE ? 'en' : process.env.BROWSERLOCALE.toLowerCase();
+}
+
+/**
+ * Returns a set of screen dimensions in numbers.
+ *
+ * This function takes the string output of `targetScreenSize` and returns an object
+ * key/value mapping of the screen diemensions represented by the string.
+ *
+ * @returns {Number, Number} Object with key/value mapping of screen dimensions.
+ */
+export function getScreenDimension() {
+	switch ( targetScreenSize() ) {
+		case 'mobile':
+			return { width: 400, height: 1000 };
+		case 'tablet':
+			return { width: 1024, height: 1000 };
+		case 'desktop':
+			return { width: 1440, height: 1000 };
+		case 'laptop':
+			return { width: 1400, height: 790 };
+		default:
+			throw new Error( 'Unsupported screen size specified.' );
+	}
+}
+
+/**
+ * Returns a new instance of a BrowserContext.
+ *
+ * A BrowserContext represents an isolated environment, akin to incognito mode
+ * inside which a single test suite is run.
+ * BrowserContexts are cheap to create and incur low overhead costs while allowing
+ * for parallelization of test suites.
+ *
+ * @returns {Promise<playwright.BrowserContext>} New BrowserContext instance.
+ */
+export async function newBrowserContext() {
+	// If no existing instance of a Browser, then launch a new instance.
+	if ( ! global.__BROWSER__ ) {
+		global.__BROWSER__ = await launchBrowser();
```

Comment:
Is it best practice in Playwright to use globals to access this stuff?

Reply:
I modeled the use of Global to store the Browser instance from the current Selenium based suite. I am definitely not certain if this is the best way to manage the browser instance, so I'm open to ideas.

Reply:
I feel like a local singleton or variable accessed through imports would be more "normal", but I guess test/NodeJS code can be a bit different.

instead of `global.__BROWSER__`, maybe we just do `let browser = null;` in the outer scope of this file, and then either export that or access it with exported functions.

It might also be a good use case for a class, since there is some local "state" to manage. Just some ideas, nothing blocking :)

Reply:
In the latest iteration I have removed and discontinue usage of `global` objects, instead using module scoped variables that are exported. Thank you for the feedback!

In the future, if it is warranted, it may make sense to extract state-related code represented by the `browser` variable to is own class. At this moment, I don't think it's necessary as the only things that need to be tracked are whether an instance of a browser exists or not.

Reply:
makes sense